File: backend/src/domain/models/materia.py
class Materia:
    def __init__(self, nome: str, id_professor: int, id_disciplina: int, id_materia: int) -> None:
        self.id = id_materia
        self.nome = nome
        self.id_disciplina = id_disciplina
        self.id_professor = id_professor

File: frontend/entites/Materia.ts
/* eslint-disable @typescript-eslint/no-explicit-any */

export const MateriaAPI = ({ nome, idDisciplina, idProfessor }: any) => ({
  nome: nome,
  id_disciplina: idDisciplina,
  id_professor: idProfessor,
});

export const MateriaInfo = (data: any) => ({
  id: data.id,
  nome: data.nome,
  idDisciplina: data.id_disciplina,
  idProfessor: data.id_professor,
});
